src/components/AdvertsTableRow/AdvertsTableRow.tsx

```diff
@@ -80,6 +83,14 @@ const AdvertsTableRow = memo((props: TAdvertsTableRowRenderer) => {
         isAdvertiserBarred ? undefined : history.push(`${ADVERTISER_URL}/${id}?currency=${localCurrency}`);
     };
 
+    useEffect(() => {
+        // If the user has become an advertiser, open the BuySellForm modal specifically for the selected advert.
+        if (hasCreatedAdvertiser && !isModalOpenFor('BuySellForm') && selectedAdvertId === advertId) {
+            showModal('BuySellForm');
+            setSelectedAdvertId(undefined);
+        }
+    }, [advertId, hasCreatedAdvertiser, hideModal, isModalOpenFor, selectedAdvertId, showModal]);
```

Comment:
do we need the hidemodal in the dependency?

Reply:
we could remove it, but i instead opted to add it to the dependency instead of adding the eslint ignore. It didn't cause any rendering issues when I had checked.

Reply:
but is it being used in the useeffect